eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -193,18 +195,33 @@ stages:
                               Commitish: $(Build.SourceVersion)
                               WorkingDirectory: $(Pipeline.Workspace)/azure-sdk-for-java
                       - template: tools/gpg/gpg.yml@azure-sdk-build-tools
-                      - template: /eng/pipelines/templates/steps/java-publishing.yml
-                        parameters:
-                          ArtifactID: ${{artifact.name}}
-                          GroupID: ${{artifact.groupId}}
-                          ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
-                          Target: MavenCentral
-                          BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
-                          JavaRepoRoot: $(Pipeline.Workspace)/azure-sdk-for-java
-                      - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
-                        parameters:
-                          ArtifactName: ${{parameters.ArtifactName}}-${{artifact.name}}-mavencentral-$(System.JobAttempt)
-                          ArtifactPath: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                      - ${{ if eq(parameters.UseEsrpRelease, 'true') }}:
+                        - template: /eng/pipelines/templates/steps/java-publishing.yml
+                          parameters:
+                            ArtifactID: ${{artifact.name}}
+                            GroupID: ${{artifact.groupId}}
+                            ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                            OutputDirectory: $(Pipeline.Workspace)/EsrpPackages
+                            Target: EsrpRelease
+                            BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
+                            JavaRepoRoot: $(Pipeline.Workspace)/azure-sdk-for-java
+                        - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
+                          parameters:
+                            ArtifactName: ${{parameters.ArtifactName}}-${{artifact.name}}-esrp-$(System.JobAttempt)
+                            ArtifactPath: $(Pipeline.Workspace)/EsrpPackages
+                      - ${{ else }}:
+                        - template: /eng/pipelines/templates/steps/java-publishing.yml
+                          parameters:
+                            ArtifactID: ${{artifact.name}}
+                            GroupID: ${{artifact.groupId}}
+                            ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
+                            Target: MavenCentral
+                            BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
+                            JavaRepoRoot: $(Pipeline.Workspace)/azure-sdk-for-java
+                        - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
```

Comment:
Do we think it is worth adding this publish artifact step in the java-publishing.yml template? It seems like we are doing these 2 steps connected in the places I see here.